File: iconex/src/main/java/foundation/icon/iconex/view/SplashActivity.java
package foundation.icon.iconex.view;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.os.Handler;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import foundation.icon.ICONexApp;
import foundation.icon.iconex.R;
import foundation.icon.iconex.dialogs.PermissionConfirmDialog;
import foundation.icon.iconex.service.VersionCheck;
import foundation.icon.iconex.util.FingerprintAuthBuilder;
import foundation.icon.iconex.util.PreferenceUtil;
import loopchain.icon.wallet.core.request.RequestData;

public class SplashActivity extends AppCompatActivity {
    private static final String TAG = SplashActivity.class.getSimpleName();

    private final int PERMISSION_REQUEST = 10001;

    private RequestData request;

    private boolean isIconConnect = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setNavigationBarColor(getResources().getColor(R.color.primary));
        setContentView(R.layout.activity_splash);

        if (getIntent() != null)
            isIconConnect = getIntent().getBooleanExtra("icon_connect", false);

        setAnimation();
    }

    @Override
    public void onResume() {
        super.onResume();

        Handler localHandler = new Handler();
        localHandler.postDelayed(new Runnable() {
            @Override
            public void run() {
                VersionCheck versionCheck = new VersionCheck(SplashActivity.this,
                        new VersionCheck.VersionCheckCallback() {
                            @Override
                            public void onNeedUpdate() {
                                // Do nothing.
                            }

                            @Override
                            public void onPass() {
                                if (isIconConnect) {
//                                    if (ICONexApp.connectMethod == Constants.Method.NONE) {
//                                        IconexConnect iconexConnect = new IconexConnect(SplashActivity.this, request);
//                                        if (ICONexApp.wallets.size() > 0)
//                                            iconexConnect.startConnectActivity();
//                                        else
//                                            IconexConnect.sendError(SplashActivity.this, request,
//                                                    new ErrorCodes.Error(ErrorCodes.ERR_EMPTY, ErrorCodes.MSG_EMTPY));
//                                    } else {
//                                        finish();
//                                    }
                                    finish();
                                } else
                                    checkPermissionConfirm();
                            }
                        });
                versionCheck.execute();
            }
        }, 500);
    }

    private void checkPermissionConfirm() {
        PermissionConfirmDialog dialog = new PermissionConfirmDialog(this);
        dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
            @Override
            public void onDismiss(DialogInterface dialog) {
                PreferenceUtil preferenceUtil = new PreferenceUtil(SplashActivity.this);
                preferenceUtil.setPermissionConfirm(true);
                preferenceUtil.loadPreference();

                startActivity();
            }
        });

        if (!ICONexApp.permissionConfirm)
            dialog.show();
        else
            startActivity();
    }

    private void startActivity() {
        if (ICONexApp.wallets.size() > 0) {
            if (ICONexApp.isLocked) {
                StartAuthenticate startAuthenticate = new StartAuthenticate();
                startAuthenticate.execute();
            } else {
                startActivity(new Intent(SplashActivity.this, MainWalletActivity.class)
                        .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK));
            }
        } else {
            startActivity(new Intent(SplashActivity.this, IntroActivity.class)
                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK));

            finish();
        }
    }

    class StartAuthenticate extends AsyncTask<Void, Void, Void> {
        @Override
        protected Void doInBackground(Void... voids) {
            if (ICONexApp.useFingerprint) {
                FingerprintAuthBuilder builder = new FingerprintAuthBuilder(SplashActivity.this);

                try {
                    boolean hasKey = builder.hasKey();

                    if (!hasKey)
                        builder.createKey(FingerprintAuthBuilder.DEFAULT_KEY_NAME, true);
                } catch (Exception e) {
                    builder.createKey(FingerprintAuthBuilder.DEFAULT_KEY_NAME, true);
                }
            }

            return null;
        }

        @Override
        protected void onPostExecute(Void aVoid) {
            super.onPostExecute(aVoid);

            startActivity(new Intent(SplashActivity.this, AuthActivity.class)
                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK));
        }
    }

    private void setAnimation() {
        ImageView imgLogo01 = findViewById(R.id.img_logo_01);
        ImageView imgLogo02 = findViewById(R.id.img_logo_02);
        TextView txtCopyright = findViewById(R.id.txt_copyright);

        Animation aniLogo01 = AnimationUtils.loadAnimation(this, R.anim.logo_01_rotation);
        Animation aniLogo02 = AnimationUtils.loadAnimation(this, R.anim.logo_02_rotation);
        Animation aniFadeInOut = AnimationUtils.loadAnimation(this, R.anim.fadeinout);

        imgLogo01.startAnimation(aniLogo01);
        imgLogo02.startAnimation(aniLogo02);
        txtCopyright.startAnimation(aniFadeInOut);
    }
}

File: iconex/src/main/java/foundation/icon/iconex/view/SplashActivity.kt
package foundation.icon.iconex.view

import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import foundation.icon.ICONexApp
import foundation.icon.connect.RequestData
import foundation.icon.iconex.R
import foundation.icon.iconex.dialogs.PermissionConfirmDialog
import foundation.icon.iconex.intro.auth.AuthActivity
import foundation.icon.iconex.util.FingerprintAuthBuilder
import foundation.icon.iconex.util.PreferenceUtil
import foundation.icon.iconex.wallet.main.MainActivity

class SplashActivity : AppCompatActivity() {
    val TAG = this@SplashActivity::class.simpleName

    private val PERMISSION_REQUEST = 10001

    private val request: RequestData? = null

    private var isIconConnect = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        if (intent != null)
            isIconConnect = intent.getBooleanExtra("icon_connect", false)
    }

    public override fun onResume() {
        super.onResume()

//        val localHandler = Handler()
//        localHandler.postDelayed({
//            val versionCheck = VersionCheck(this@SplashActivity,
//                    object : VersionCheck.VersionCheckCallback {
//                        override fun onNeedUpdate() {
//                            // Do nothing.
//                        }
//
//                        override fun onPass() {
//                            if (isIconConnect) {
//                                finish()
//                            } else
//                                checkPermissionConfirm()
//                        }
//                    })
//            versionCheck.execute()
//        }, 500)

        startActivity(Intent(this@SplashActivity, CreateActivity::class.java))
    }

    private fun checkPermissionConfirm() {
        val dialog = PermissionConfirmDialog(this, R.style.AppTheme)
        dialog.setOnDismissListener {
            val preferenceUtil = PreferenceUtil(this@SplashActivity)
            preferenceUtil.setPermissionConfirm(true)

            startActivity()
        }

        if (!ICONexApp.permissionConfirm)
            dialog.show()
        else
            startActivity()
    }

    private fun startActivity() {
        if (ICONexApp.mWallets.size > 0) {
            if (ICONexApp.isLocked) {
                val startAuthenticate = StartAuthenticate()
                startAuthenticate.execute()
            } else {
                startActivity(Intent(this@SplashActivity, MainActivity::class.java)
                        .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK))
            }
        } else {
            startActivity(Intent(this@SplashActivity, IntroActivity::class.java)
                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK))

            finish()
        }
    }

    internal inner class StartAuthenticate : AsyncTask<Void, Void, Void>() {
        override fun doInBackground(vararg voids: Void): Void? {
            if (ICONexApp.useFingerprint) {
                val builder = FingerprintAuthBuilder(this@SplashActivity)

                try {
                    val hasKey = builder.hasKey()

                    if (!hasKey)
                        builder.createKey(FingerprintAuthBuilder.DEFAULT_KEY_NAME, true)
                } catch (e: Exception) {
                    builder.createKey(FingerprintAuthBuilder.DEFAULT_KEY_NAME, true)
                }

            }

            return null
        }

        override fun onPostExecute(aVoid: Void) {
            super.onPostExecute(aVoid)

            startActivity(Intent(this@SplashActivity, AuthActivity::class.java)
                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK))
        }
    }
}
